File: tests/semantic_gold_fixtures/python/clean_complex.py
def clean_parse_tokens(text):
    tokens = []
    current = []
    quoted = False
    escaped = False
    for char in text:
        if escaped:
            current.append(char)
            escaped = False
        elif char == "\\":
            current.append(char)
            escaped = True
        elif char == '"':
            current.append(char)
            quoted = not quoted
        elif char.isspace() and not quoted:
            if current:
                tokens.append("".join(current))
                current = []
        else:
            current.append(char)
    if current:
        tokens.append("".join(current))
    return tokens


def clean_validate_contract(result):
    if not isinstance(result, dict):
        raise ValueError("result must be an object")
    tier = result.get("tier")
    smelly = result.get("smelly")
    if smelly != (tier != "clean"):
        raise ValueError("tier and smelly disagree")
    return tier


def clean_extract_json_payload(content, parse, recover):
    try:
        return parse(content)
    except ValueError:
        pass
    return recover(content)


def kinda_forward_payload(payload):
    forwarded = payload
    return forwarded


def unresolved_external_boundary(value):
    return external_boundary(value)

File: tests/semantic_gold_fixtures/typescript/clean_complex.ts
export function clean_parse_tokens(text: string): string[] {
  const tokens: string[] = [];
  let current = "";
  let quoted = false;
  let escaped = false;
  for (const char of text) {
    if (escaped) {
      current += char;
      escaped = false;
    } else if (char === "\\") {
      current += char;
      escaped = true;
    } else if (char === '"') {
      current += char;
      quoted = !quoted;
    } else if (/\s/.test(char) && !quoted) {
      if (current) {
        tokens.push(current);
        current = "";
      }
    } else {
      current += char;
    }
  }
  if (current) tokens.push(current);
  return tokens;
}

export function clean_validate_contract(result: { tier: string; smelly: boolean }): string {
  if (!result || typeof result !== "object") {
    throw new Error("result must be an object");
  }
  const tier = result.tier;
  const smelly = result.smelly;
  if (smelly !== (tier !== "clean")) {
    throw new Error("tier and smelly disagree");
  }
  return tier;
}

export function kindaForwardPayload(payload: Payload): Payload {
  const forwarded = payload;
  return forwarded;
}
